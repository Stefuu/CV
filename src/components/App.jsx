import React from 'react'
import styled, { ThemeProvider } from 'styled-components'
import theme from '../components/themes/default'
import Flex from '../components/elements/Flex'
import Header from '../components/Header'
import Xperience from '../components/Xperience'
import Education from '../components/Education'
import Skills from '../components/Skills'
import Languages from '../components/Languages'
import data from '../static/data/data.json'

const Container = styled(Flex)`
    margin: 0 auto;
    border-radius: 8px;
    padding: 25px;
    font-family: Lato;
`

const App = () => {
  if (!data) return null

  return (
    <ThemeProvider theme={theme}>
      <Container
        maxWidth={[600, 600, 800]}>
        <Flex
          alignItems={['center', 'flex-start']}
          width='100%'
          flexDirection='column'>
          <Header {...data.header} />
          <Xperience {...data.xperience} />
          <Education {...data.education} />
          <Skills skills={data.skills} />
          <Languages {...data.languages} />
        </Flex>
      </Container>
    </ThemeProvider>
  )
}

export default App
