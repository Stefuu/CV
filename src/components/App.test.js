import React from 'react'
import { render } from 'react-testing-library'
import App from './App'

describe('App', () => {
  it('Deve renderizar [snapshot]', async () => {
    const { container } = render(<App />)
    expect(container.firstChild).toMatchSnapshot()
  })
})
