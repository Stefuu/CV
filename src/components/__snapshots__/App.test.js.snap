// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Deve renderizar [snapshot] 1`] = `
<div
  class="sc-ifAKCX kkLZEE sc-bdVaJa hkTqNZ"
>
  <div
    class="sc-bdVaJa hudVDw"
  >
    <div
      class="sc-bdVaJa hilsGc"
    >
      <div
        class="sc-bdVaJa dnIFYn"
      >
        <img
          alt="Stéfano"
          class="sc-bwzfXH dmnngk"
          src="me.jpeg"
        />
      </div>
      <div
        class="sc-bdVaJa hvxHyA"
      >
        <div
          class="sc-bdVaJa heyuGm"
        >
          <strong>
            Stéfano Damiano
          </strong>
          <div
            class="sc-bdVaJa cLkge"
          >
            <strong>
              Desenvolvedor Front-End
            </strong>
          </div>
        </div>
        <div
          class="sc-bdVaJa jtVCEF"
        >
          <div
            class="sc-bdVaJa ccJYFI"
          >
            <div
              class="sc-bdVaJa gsLGXC"
              style="white-space: nowrap;"
            >
              <span
                style="font-weight: bold;"
              >
                Phone
                : 
              </span>
              <span>
                11 97027-9708
              </span>
            </div>
          </div>
          <div
            class="sc-bdVaJa ccJYFI"
          >
            <div
              class="sc-bdVaJa gsLGXC"
              style="white-space: nowrap;"
            >
              <span
                style="font-weight: bold;"
              >
                Nasc
                : 
              </span>
              <span>
                08/08/1989
              </span>
            </div>
          </div>
          <div
            class="sc-bdVaJa ccJYFI"
          >
            <div
              class="sc-bdVaJa gsLGXC"
              style="white-space: nowrap;"
            >
              <span
                style="font-weight: bold;"
              >
                Cidade
                : 
              </span>
              <span>
                São Paulo - SP
              </span>
            </div>
          </div>
          <div
            class="sc-bdVaJa ccJYFI"
          >
            <div
              class="sc-bdVaJa gsLGXC"
              style="white-space: nowrap;"
            >
              <span
                style="font-weight: bold;"
              >
                Email
                : 
              </span>
              <span>
                mineiro.guaxupe@gmail.com
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa iDJurw"
    >
      <div
        class="sc-bdVaJa dBCrBP"
      >
        <strong>
          Experiências:
        </strong>
      </div>
      <div
        class="sc-bdVaJa jfEzKd"
      >
        <div
          class="sc-bdVaJa UqLRs"
        >
          <div
            class="sc-bdVaJa gGJTwr"
          >
            <strong>
              - TOVS (idexo)
            </strong>
          </div>
          <div
            class="sc-bdVaJa hOejgn"
          >
            <div
              class="sc-bdVaJa cZHmfT"
            >
              <span>
                De: Jan/2018 até Hoje
              </span>
            </div>
            <div
              class="sc-bdVaJa nLDLh"
            >
              <span>
                Cargo: Desenvolvedor Front-End Sênior
              </span>
            </div>
            <div
              class="sc-bdVaJa iSxgmk"
            >
              <span>
                Descrição: Criação e manutenção de um e-commerce capaz de se adequar a regras de negócio complexas de produtos da empresa
              </span>
            </div>
            <div
              class="sc-bdVaJa eCYppH"
            >
              <div
                class="sc-bdVaJa hvxHyA"
              >
                <span>
                  Destaques:
                </span>
              </div>
              <ul
                class="sc-htpNat datbWf"
              >
                <li
                  class="sc-bdVaJa cWwZGa"
                >
                  - Ambiente de inovação com intenção de renovar a cultura da velha TOTVS. Trabalhamos lado a lado com start-ups incubadas pela TOTVS, o que permite agregar muito sobre tecnologias e soluções muito quentes
                </li>
                <li
                  class="sc-bdVaJa LGxXB"
                >
                  - Tivemos a oportunidade de começar tudo do zero, portanto escolhemos a stack, que é: React com styled-components, styled-system, fazendo uso de context API, Graphql com Relay modern, Node (express) e MQTT (para mensagem entre micro-serviços)
                </li>
                <li
                  class="sc-bdVaJa LGxXB"
                >
                  - Criação de uma arquitetura de execução de regras de negócio no front-end que não são fixas no mesmo. Usamos uma estrutura declarativa com https://github.com/jwadhams/json-logic-js que permite que as regras sejam geradas pelo back-end e apénas consumidas e executadas pelo front (evitando um oceano de condições no front).
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="sc-bdVaJa UqLRs"
        >
          <div
            class="sc-bdVaJa gGJTwr"
          >
            <strong>
              - Universo Online (UOL)
            </strong>
          </div>
          <div
            class="sc-bdVaJa hOejgn"
          >
            <div
              class="sc-bdVaJa cZHmfT"
            >
              <span>
                De: Jul/2012 até Dez/2017
              </span>
            </div>
            <div
              class="sc-bdVaJa nLDLh"
            >
              <span>
                Cargo: Desenvolvedor Front-End Pleno
              </span>
            </div>
            <div
              class="sc-bdVaJa iSxgmk"
            >
              <span>
                Descrição: Criação e manutenção de páginas do portal (layout, integrações com back-end e sistemas externos), criação e manutenção de sistemas internos (principalmente em node), criação de serviços
              </span>
            </div>
            <div
              class="sc-bdVaJa eCYppH"
            >
              <div
                class="sc-bdVaJa hvxHyA"
              >
                <span>
                  Destaques:
                </span>
              </div>
              <ul
                class="sc-htpNat datbWf"
              >
                <li
                  class="sc-bdVaJa cWwZGa"
                >
                  - Passei por diversas areas de desenvolvimento da empresa, ganhando bom conhecimento das dificuldades e desafios de colocar e manter um portal com milhões de acessos em produção.
                </li>
                <li
                  class="sc-bdVaJa LGxXB"
                >
                  - Passei pelas dificuldades de se criar layouts responsivos cross-browser que precisam rodar em navegadores antigos.
                </li>
                <li
                  class="sc-bdVaJa LGxXB"
                >
                  - Desenvolvi integrações com ferramentas experimentais do google e facebook. Como o UOL tinha parceria com essas empresas, e era um bom laboratório pela quantidade de acessos, eles liberavam algumas ferramentas novas antes do lançamento. Ferramentas como AMP, Instant Articles e Facebook Messenger BOTS.
                </li>
                <li
                  class="sc-bdVaJa LGxXB"
                >
                  - Aprendi muito sobre arquitetura de back-end trabalhando na criação do micro-serviço do Facebook Messenger BOT, pelo fato de ser necessário fazer envio de milhões de mensagens (quando aparecia alguma notícia bombástica), aprendi muito sobre como escalar uma aplicação node, kubernetes, sobre uso de filas (rabbitmq principalmente) e banco de dados não relacional (mongodb)
                </li>
              </ul>
            </div>
          </div>
        </div>
        <div
          class="sc-bdVaJa UqLRs"
        >
          <div
            class="sc-bdVaJa gGJTwr"
          >
            <strong>
              - KAITS - Sistema educacional
            </strong>
          </div>
          <div
            class="sc-bdVaJa hOejgn"
          >
            <div
              class="sc-bdVaJa cZHmfT"
            >
              <span>
                De: Fev/2011 até Jan/2012
              </span>
            </div>
            <div
              class="sc-bdVaJa nLDLh"
            >
              <span>
                Cargo: Estagiário
              </span>
            </div>
            <div
              class="sc-bdVaJa iSxgmk"
            >
              <span>
                Descrição: Atuação com javascript vanilla na construção e manutenção de um sistema web para escolas de idioma
              </span>
            </div>
            <div
              class="sc-bdVaJa eCYppH"
            >
              <div
                class="sc-bdVaJa hvxHyA"
              >
                <span>
                  Destaques:
                </span>
              </div>
              <ul
                class="sc-htpNat datbWf"
              >
                <li
                  class="sc-bdVaJa cWwZGa"
                >
                  -Uso de javascript sem nenhum framework foi uma experiência interessante, principalmente por estar começando a usar js na época
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa bGLETG"
    >
      <div
        class="sc-bdVaJa dbRsal"
      >
        <strong>
          Educação:
        </strong>
      </div>
      <div
        class="sc-bdVaJa hHKtwu"
      >
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <strong>
            Universidade Presbiteriana Mackenzie
          </strong>
        </div>
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <span>
            Bacharel em Ciência da Computação
          </span>
        </div>
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <span>
            2008 - 2012
          </span>
        </div>
      </div>
    </div>
    <div
      class="sc-bdVaJa bGLETG"
    >
      <div
        class="sc-bdVaJa dbRsal"
      >
        <strong>
          Skills:
        </strong>
      </div>
      <ul
        class="sc-bdVaJa sc-bxivhb jgcSTk"
      >
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Javascript
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            ES20XX
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            HTML5
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            CSS3
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            React
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Graphql
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Relay
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Unity Test
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Jest
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Enzyme
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Nightwatch
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Linux
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            GIT
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Docker
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Gulp / Grunt
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Node
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Mongodb
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            Rabbitmq
          </span>
        </li>
        <li
          class="sc-bdVaJa hEZfMU"
          font-size="15,18"
        >
          <span>
            - 
            SCRUM
          </span>
        </li>
      </ul>
    </div>
    <div
      class="sc-bdVaJa bGLETG"
    >
      <div
        class="sc-bdVaJa dbRsal"
      >
        <strong>
          Idiomas:
        </strong>
      </div>
      <div
        class="sc-bdVaJa hHKtwu"
      >
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <strong>
            Português
          </strong>
          <span>
             - 
            Nativo
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa hHKtwu"
      >
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <strong>
            Inglês
          </strong>
          <span>
             - 
            Intermediário
          </span>
        </div>
      </div>
      <div
        class="sc-bdVaJa hHKtwu"
      >
        <div
          class="sc-bdVaJa hvxHyA"
        >
          <strong>
            Espanhol
          </strong>
          <span>
             - 
            Básico
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
